File: BigCupPayroll/ClientApp/src/app/models/payroll.ts
export interface IPayroll {
    Id: number;
    UserId: string;
    Employee: string;
    GrossPay: number;
    NetPay: number;
    CutOffDate:Date;  
    PresentDay:number;  
    BasicPay:number;  
    allowances: IAllowance[],
    deductions: IDeduction[]
}

export interface IAllowance {
    Description: string,
    Amount: string,
    PayrollId: number
  }

export interface IDeduction {
Description: string,
Amount: string,
PayrollId: number
}

File: CommonLibrary/Models/Payroll.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CommonLibrary.Models
{
    public class Payroll
    {
        public int Id { get; set; }
        public DateTime CutOffDate { get; set; }
        public decimal GrossPay { get; set; }
        public decimal NetPay { get; set; }
        public int UserId { get; set; }
        public string Employee { get; set; }
        public decimal BasicPay { get; set; }
        public int PresentDay { get; set; }
        public List<Allowance> allowances { get; set; }
        public List<Deduction> deductions { get; set; }

    }

    public class AdjustedPayroll
    {
        public Payroll newPay { get; set; }
        public List<Allowance> adjustment { get; set; }
    }
}
